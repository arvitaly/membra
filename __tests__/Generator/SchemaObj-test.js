"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const schema2_1 = require("./../../__fixtures__/schema2");
const SchemaObj_1 = require("./../../Generator/SchemaObj");
describe("SchemaObj", () => {
    const executor = ({ query: queryObj }) => {
        const m1 = queryObj.viewer.model1({
            first: 10,
        }).edges.map(({ node }) => {
            return {
                id: node.id,
                field1: node.field1,
                model2: node.model2({ where: { field1: ["test"] } }).id,
            };
        });
        const m2 = queryObj.viewer.model1({
            first: 15,
        }).edges.map(({ node }) => {
            return node.model2({ where: { field1: ["test5"] } }).field2;
        });
        return { m1, m2 };
    };
    it("get query", () => {
        const schemaObj = new SchemaObj_1.default(schema2_1.default);
        executor(schemaObj);
        expect(schemaObj.getQuery()).toMatchSnapshot();
    });
    it("fill data", () => {
        const schemaObj = new SchemaObj_1.default(schema2_1.default);
        expect(schemaObj.fillData({
            viewer: {
                f1: {
                    edges: [{
                            node: {
                                id: "ID22",
                                field1: "newValue",
                                model2: {
                                    field2: "Field2valueSame",
                                },
                            },
                        }],
                },
                model1: {
                    edges: [{
                            node: {
                                id: "ID11",
                                field1: "Test",
                                model2: {
                                    id: "ID56",
                                    field2: "Field2value",
                                },
                            },
                        }],
                },
            },
        }, executor)).toMatchSnapshot();
    });
});
