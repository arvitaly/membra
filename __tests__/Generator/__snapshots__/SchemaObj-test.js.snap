// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchemaObj fill data 1`] = `
Object {
  "m1": Array [
    Object {
      "field1": "Test",
      "id": "ID11",
      "model2": "ID56",
    },
  ],
  "m2": Array [
    "Field2valueSame",
  ],
}
`;

exports[`SchemaObj get query 1`] = `
"query {
    viewer {
        model1(first: 10) {
            edges {
                node {
                    id
                    field1
                    model2(where: {field1: [\\"test\\"], nullableField: null, field3: [field4: \\"field4Value1\\"]}) {
                        id
                    }
                }
            }
        }
        f1:model1(first: 15) {
            edges {
                node {
                    model2(where: {field1: [\\"test5\\"], field3: [field4: \\"field4Value2\\"]}) {
                        field2
                    }
                }
            }
        }
    }
}"
`;
